Lib/test/test_contextlib_async.py

```diff
@@ -278,6 +278,25 @@ async def woohoo(self, func, args, kwds):
         async with woohoo(self=11, func=22, args=33, kwds=44) as target:
             self.assertEqual(target, (11, 22, 33, 44))
 
+    async def test_recursive(self):
```

Comment:
should it at least be called `test_decorator`?

But for the non-async case there are extensive tests for ContextDecorator.  They should be reviewed and copied where relevant.

https://github.com/python/cpython/blob/858a98fbe9446e4ec781b7f3caaf90505ce2ca23/Lib/test/test_contextlib.py#L395-L552

Reply:
Yes, forgot this, thanks.